tests/Behat/FeatureContext.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_editorial\Behat;
+
+use Drupal\DrupalExtension\Context\RawDrupalContext;
+use Behat\Mink\Exception\ExpectationException;
+use Behat\Mink\Element\NodeElement;
+use Behat\Gherkin\Node\TableNode;
+use PHPUnit\Framework\Assert;
+
+/**
+ * Defines step definitions that are generally useful in this project.
+ */
+class FeatureContext extends RawDrupalContext {
+
+  /**
+   * Checks that the given select field has the options listed in the table.
+   *
+   * @Then I should have the following options for :select select:
+   */
+  public function assertSelectOptions($select, TableNode $options): void {
+    // Retrieve the specified field.
+    if (!$field = $this->getSession()->getPage()->findField($select)) {
+      throw new ExpectationException("Field '$select' not found.", $this->getSession());
+    }
+
+    // Check that the specified field is a <select> field.
+    $this->assertElementType($field, 'select');
+
+    // Retrieve the options table from the test scenario and flatten it.
+    $expected_options = $options->getRows();
+    array_walk($expected_options, function (&$value) {
+      $value = reset($value);
+    });
+
+    // Retrieve the actual options that are shown in the page.
+    $actual_options = $field->findAll('css', 'option');
+
+    // Convert into a flat list of option text strings.
+    array_walk($actual_options, function (&$value) {
+      $value = $value->getText();
+    });
+
+    // Check that all expected options are present.
+    foreach ($expected_options as $expected_option) {
+      if (!in_array($expected_option, $actual_options)) {
+        throw new ExpectationException("Option '$expected_option' is missing from select list '$select'.", $this->getSession());
+      }
+    }
+  }
+
+  /**
+   * Checks that the given element is of the given type.
+   *
+   * @param \NodeElement $element
+   *   The element to check.
+   * @param string $type
+   *   The expected type.
+   *
+   * @throws \ExpectationException
+   *   Thrown when the given element is not of the expected type.
+   */
+  public function assertElementType(NodeElement $element, $type): void {
```

Comment:
This should be a protected method. And I would move it to the bottom. We don't have a rule but I think it's neat when all the public methods are at the top (steps) and helper protected methods at the bottom.